### ISSUE TITLE
Readme needs update

### ISSUE THREAD

**Issue by jafermarq** [ARMmbed/mbed-os-example-filesystem#81]
I believe the Readme, in particular, the `changing-the-block-device` section, needs to be updated after the latest change to `main.cpp`. Also, shouldn't the SPI numbering be specified in the `mbed_app.json`. Although I'm able to compile this project, executing the generated binary will result in a hard fault.

**Comment by dannybenor** [comment]
@jafermarq can you provide more specific info? What target are you using? Did you change anything?

**Comment by jafermarq** [comment]
I'm using a Nucleo-F767ZI and `mbed-cli 1.8.3`. I can compile the example in `master` but when flashing it, it will result in a HardFault when attempting to mount the filesystem. This error seems to trigger a reset so this is happening in a loop over and over. The error code is `0x800259E` By switching to `override-storage-components` branch I can make it work. I'm mostly interested in using a SD card, so I haven't tested other types of `BlockDevice` interfaces.

**Comment by dannybenor** [comment]
@jafermarq Did you add support for Nucleo-F767ZI in targets.json? Can you paste here your compilation command?

**Comment by jafermarq** [comment]
@dannybenor, I haven't edited `targets.json`. I run the following command: `mbed compile -m auto -t GCC_ARM`.

**Comment by RonEld** [comment]
Hi @jafermarq 
What is set as you component in your target?
Did you add
~~~
"target_overrides": {
         ...
         "NUCLEO_F767ZI": {
             "target.components_add": ["SD"],
             ...
         },
         ...
  }
~~~
to your `mbed_app.json` ?

**Comment by linlingao** [comment]
@jafermarq Can you respond to @RonEld? If you've solved your issue, please close.

**Comment by ciarmcom** [comment]
Thank you for raising this issue. Please note we have updated our policies and 
 now only defects should be raised directly in GitHub. Going forward questions and 
 enhancements will be considered in our forums, https://forums.mbed.com/ . If this 
 issue is still relevant please re-raise it there.
 This GitHub issue will now be closed.

**Comment by adbridge** [comment]
Apologies, this was closed in error by the automated bot - re-opening.

**Comment by ciarmcom** [comment]
Thank you for raising this detailed GitHub issue. I am now notifying our internal issue triagers. 
*Internal Jira reference: https://jira.arm.com/browse/IOTOSM-2068*

**Comment by evedon** [comment]
Fixed by #133
